src/http_server.cc

```diff
@@ -3078,154 +3077,262 @@ HTTPAPIServer::HandleGenerate(
+  RETURN_AND_RESPOND_IF_ERR(
+      req, TRITONSERVER_InferenceRequestNew(
+               &irequest, server_.get(), model_name.c_str(),
+               requested_model_version));
 
-  // TODO: Look for well-defined keys and reject if missing required keys
-  for (const auto& m : members) {
-    LOG_INFO << "TOP LEVEL REQUEST MEMBER: " << m;
-    if (m == "prompt") {
-      RETURN_AND_RESPOND_IF_ERR(
-          req, request.MemberAsString(m.c_str(), &prompt));
-      LOG_INFO << "FOUND PROMPT: " << prompt;
-    } else if (m == "parameters") {
-      LOG_INFO << "FOUND PARAMETERS";
-      RETURN_AND_RESPOND_IF_ERR(
-          req, request.MemberAsObject(m.c_str(), &params_json));
-      // [Debug] Parse parameters
-      std::vector<std::string> param_keys;
-      // TODO: Improve error message for bad json objects like RETURN_MSG_IF_ERR
-      RETURN_AND_RESPOND_IF_ERR(req, params_json.Members(&param_keys));
-      for (const auto& pk : param_keys) {
-        LOG_INFO << "FOUND PARAMETER KEY: " << pk;
+
+  // HTTP request paused when creating inference request. Resume it on exit if
+  // this function returns early due to error. Otherwise resumed in callback.
+  std::unique_ptr<GenerateRequestClass> generate_request{
+          new GenerateRequestClass(server_.get(), req, GetResponseCompressionType(req),
+          generate_response_schema_.get(),
+          streaming /* server-sent events */, irequest)};
+
+  const char* request_id = "<id_unknown>";
+  // Callback to cleanup on any errors encountered below. Capture everything
+  // by reference to capture local updates, except for shared pointers which
+  // should be captured by value in case of ref count issues.
+  auto error_callback = [&](TRITONSERVER_Error* error) {
+    if (error != nullptr) {
+      // Get request ID for logging in case of error.
+      if (irequest != nullptr) {
+        LOG_TRITONSERVER_ERROR(
+            TRITONSERVER_InferenceRequestId(irequest, &request_id),
+            "unable to retrieve request ID string");
+      }
+      if (!strncmp(request_id, "", 1)) {
+        request_id = "<id_unknown>";
       }
-    } else {
-      LOG_INFO << "FOUND UNEXPECTED KEY: " << m;
-    }
-  }
 
-  // Form Triton tensor-based request from simple JSON format
-  triton::common::TritonJson::Value triton_json(
-      triton::common::TritonJson::ValueType::OBJECT);
+      LOG_VERBOSE(1) << "[request id: " << request_id << "] "
+                     << "Infer failed: " << TRITONSERVER_ErrorMessage(error);
+      LOG_TRITONSERVER_ERROR(
+          TRITONSERVER_InferenceRequestDelete(irequest),
+          "deleting HTTP/REST inference request");
 
-  // TODO: Parse Model Config for candidate input names/keys to compare and
-  // populate
-  triton::common::TritonJson::Value config_json;
-  RETURN_AND_RESPOND_IF_ERR(
-      req, config_json.Parse(config_json_str.c_str(), config_json_str.size()));
+      evhtp_request_resume(req);
+      RETURN_AND_RESPOND_IF_ERR(req, error);
```

Comment:
This will double delete the error. Can return msg directly instead and let `RETURN_CALLBACK_IF_ERR` delete the error:
```cpp
      evhtp_request_resume(req);
      EVBufferAddErrorJson(req->buffer_out, error);
      evhtp_send_reply(req, EVHTP_RES_BADREQ);
```